dali/kernels/imgproc/flip_gpu_test.cu

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/imgproc/flip_gpu.cuh"
+#include <gtest/gtest.h>
+#include <random>
+#include <vector>
+#include "dali/kernels/imgproc/flip_test.h"
+#include "dali/pipeline/data/tensor.h"
+#include "dali/kernels/test/test_tensors.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+
+namespace dali {
+namespace kernels {
+
+class FlipGpuTest: public testing::TestWithParam<std::array<int64, 4>> {
+ public:
+  FlipGpuTest()
+  : tensor_shape(GetParam())
+  , _volume(volume(tensor_shape))
+  , shape({tensor_shape, tensor_shape, tensor_shape, tensor_shape,
+           tensor_shape, tensor_shape, tensor_shape, tensor_shape}) {}
+
+  void SetUp() override {
+    ttl_in.reshape(shape);
+    auto tlv = ttl_in.cpu(nullptr);
+    std::mt19937_64 rng;
+    UniformRandomFill(tlv, rng, 0., 10.);
+  }
+
+ protected:
+  std::vector<int32> flip_x{0, 0, 1, 1, 0, 0, 1, 1};
```

Comment:
Why int32 and not just int? We don't care about specific width here...

Reply:
FYI: All integer arguments are passed to the operators as int64_t and you can use any integral type in GetArgument.

Reply:
...I don't know about ArgumentInputs, however...